_docs/monolithic.md

```diff
@@ -0,0 +1,97 @@
+---
+layout: architecture
+title: Monolithic Master
+date: 2019-07-09
+version: v0.0.1
+summary: A simple master/agent architecture with all services running on one master.
+---
+
+## Intended Audience
+
+This architecture is intended for small to medium static infrastructures and is
+easier than multiple compile masters to set up and maintain. Generally speaking,
+redundancy and HA are most useful for dynamic environments. A stopped Puppet
+master doesn't prevent the existing configuration from being enforced but it will
+halt any new deployments or configuration updates. If those constraints fit your
+needs, then this architecture is suggested.
+
+
+<div class="mermaid">
+  graph LR;
+  git(Git Repository)
+
+  subgraph master[Puppet Master node]
+    Foreman(The Foreman)
+    Webhook(Puppet Webhook Server)
+    PuppetDB
+    PuppetServer{Puppet Server}
+  end
+
+  Agent1(Agent 1)
+  Agent2(Agent 2)
+  Agents(Agents ...)
+  Agent_n(Agent n)
+
+  click Foreman "https://www.theforeman.org" "Foreman is a complete lifecycle management tool for physical and virtual servers."
+  click Webhook "https://github.com/voxpupuli/puppet_webhook" "A webhook service that can trigger code deploys from source code repository updates."
+
+  git --webhook--> Webhook
+  Webhook --r10k code deploy--> PuppetServer
+
+  PuppetDB --- PuppetServer
+  Foreman --- PuppetServer
+
+  PuppetServer --> Agent1
+  PuppetServer --> Agent2
+  PuppetServer --> Agents
+  PuppetServer --> Agent_n
+</div>
+
+## Setup and Usage
+
+{write a guide on how to deploy, configure, and use this architecture}
+
+
+### Git Repository
+
+We recommend organizing your code as a Control Repository with branches for
+environments. See the [reference repository](https://github.com/puppetlabs/control-repo)
+for an example.
+
+
+### Foreman
+
+[Foreman](https://www.theforeman.org) is a complete lifecycle management tool
+for physical and virtual servers. It will provide you with a graphical
+classifier, a Hiera data source, and report monitoring. It also includes the
+power to easily automate repetitive tasks, quickly deploy applications, and
+proactively manage servers, on-premise or in the cloud.
+
+
+### Puppet Webhook
+
```

Comment:
In this section we probably should point to puppet module [puppetwebhook](https://github.com/voxpupuli/puppet-puppetwebhook/.)